client/rpc/src/chain/mod.rs

```diff
@@ -152,34 +152,22 @@ where
 	pub fn into_rpc_module(self) -> Result<RpcModule<Self>, JsonRpseeError> {
 		let mut rpc_module = RpcModule::new(self);
 
-		rpc_module.register_method("chain_getHeader", |params, chain| {
-			log::info!("chain_getBlock [{:?}]", params);
-			// TODO: make is possible to register async methods on jsonrpsee servers.
-			//https://github.com/paritytech/jsonrpsee/issues/291
-			//
-			// NOTE(niklasad1): will block the connection task on the server.
+		rpc_module.register_async_method("chain_getHeader", |params, chain| {
 			let hash = params.one().ok();
-			futures::executor::block_on(chain.header(hash)).map_err(rpc_err)
+			async move { chain.header(hash).await.map_err(rpc_err) }.boxed()
```

Comment:
dq: what is stopping this from being just `chain.header(hash)`? Isn't that a future already? The wrong kind?

Reply:
```
error[E0515]: cannot return value referencing function parameter `chain`
   --> client/rpc/src/chain/mod.rs:157:4
    |
157 |             chain.header(hash).map_err(rpc_err).boxed()
    |             -----^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |             |
    |             returns a value referencing data owned by the current function
    |             `chain` is borrowed here

```